src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
+    const payload = JSON.stringify(request.body);
+
+    if (!signature) {
+      throw new UnauthorizedException('Missing signature');
+    }
+
+    const hmac = crypto.createHmac('sha256', secret);
+    const digest = `sha256=${hmac.update(payload).digest('hex')}`;
+
+    if (!crypto.timingSafeEqual(Buffer.from(digest), Buffer.from(signature))) {
+      throw new UnauthorizedException('Invalid signature');
+    }
+
+    return true;
+  }
```

Comment:
PullProofAI

The implementation of the `WebhookSecretGuard` looks good overall. However, there are a few suggestions for improvement:

1. Consider adding comments to explain the purpose of critical sections of the code for better readability and maintainability.
2. Ensure error handling is robust and covers all possible edge cases.
3. It's recommended to validate the `request.body` exists before stringifying it to prevent potential runtime errors.

```diff
+     // Add comments to explain key sections of the code
+ 
+     // Validate request body existence before stringifying
+     if (!request.body) {
+         throw new UnauthorizedException('Request body is missing');
+     }
```

### Summary
The `WebhookSecretGuard` implementation is mostly well-done. Adding comments for clarity and validating the existence of `request.body` before using it would enhance the code further.


<!-- This is an auto-generated comment by PullProofAI -->